e2e-test-server/src/index.ts

```diff
@@ -102,13 +61,124 @@ async function pubSubPull(): Promise<void> {
   );
 }
 
-async function main(): Promise<void> {
+/**
+ * Starts an HTTP server to receive pubsub push requests. This is used for serverless
+ * environments where we can't subscribe via pull.
+ */
+function pubSubPush() {
+  if (constants.PUSH_PORT === undefined) {
+    throw new Error('PUSH_PORT must be set when in push subscription mode');
+  }
+
+  const app = express();
+
+  app
+    .use(express.json())
+    .post('/', (req, res) => {
```

Comment:
do we need to handle /ready and /alive as well?

Reply:
Didn't know those are needed, what's it for?

Reply:
Ah, might just be for app-engine.

Reply:
We can add it for that later if needed.

Reply:
Ok, can add those when I add App Engine 👍